src/app/pages/MarginTradePage/hooks/useSimulatedTrade.ts

```diff
@@ -0,0 +1,158 @@
+import { useMemo } from 'react';
+import { bignumber } from 'mathjs';
+import { useFilterSimulatorResponseLogs } from 'app/hooks/simulator/useFilterSimulatorResponseLogs';
+import { useSimulator } from 'app/hooks/simulator/useSimulator';
+import { usePositionLiquidationPrice } from 'app/hooks/trading/usePositionLiquidationPrice';
+import { Asset } from 'types';
+import { TradeEventData } from 'types/active-loan';
+import { TradingPosition } from 'types/trading-position';
+import {
+  getContract,
+  getLendingContractName,
+} from 'utils/blockchain/contract-helpers';
+import {
+  MAINTENANCE_MARGIN,
+  TRADE_LOG_SIGNATURE_HASH,
+} from 'utils/classifiers';
+
+const TradeLogInputs = [
+  {
+    indexed: true,
+    internalType: 'address',
+    name: 'user',
+    type: 'address',
+  },
+  {
+    indexed: true,
+    internalType: 'address',
+    name: 'lender',
+    type: 'address',
+  },
+  {
+    indexed: true,
+    internalType: 'bytes32',
+    name: 'loanId',
+    type: 'bytes32',
+  },
+  {
+    indexed: false,
+    internalType: 'address',
+    name: 'collateralToken',
+    type: 'address',
+  },
+  {
+    indexed: false,
+    internalType: 'address',
+    name: 'loanToken',
+    type: 'address',
+  },
+  {
+    indexed: false,
+    internalType: 'uint256',
+    name: 'positionSize',
+    type: 'uint256',
+  },
+  {
+    indexed: false,
+    internalType: 'uint256',
+    name: 'borrowedAmount',
+    type: 'uint256',
+  },
+  {
+    indexed: false,
+    internalType: 'uint256',
+    name: 'interestRate',
+    type: 'uint256',
+  },
+  {
+    indexed: false,
+    internalType: 'uint256',
+    name: 'settlementDate',
+    type: 'uint256',
+  },
+  {
+    indexed: false,
+    internalType: 'uint256',
+    name: 'entryPrice',
+    type: 'uint256',
+  },
+  {
+    indexed: false,
+    internalType: 'uint256',
+    name: 'entryLeverage',
+    type: 'uint256',
+  },
+  {
+    indexed: false,
+    internalType: 'uint256',
+    name: 'currentLeverage',
+    type: 'uint256',
+  },
+];
+
+export const useSimulatedTrade = (
+  loanToken: Asset,
+  collateral: Asset,
+  txArgs: string[],
+  position: TradingPosition,
+  amount: string,
+) => {
+  const contractName = getLendingContractName(loanToken);
+
+  const simulator = useFilterSimulatorResponseLogs<TradeEventData>(
+    useSimulator(
+      contractName,
+      'marginTrade',
+      txArgs,
+      collateral === Asset.RBTC ? amount : '0',
+      amount !== '0' &&
+        !!contractName &&
+        !!position &&
+        !txArgs.filter(item => item === '').length,
```

Comment:
```suggestion
      amount !== '0' &&
      !txArgs.includes(item => item === ''),
```

Reply:
The condition is unnecessarily complicated - `contractName` and `position` are always defined so there is no need to check if they exist.

Regarding `!txArgs.filter(item => item === '').length` - I think (and correct me if I am wrong) you want to make sure there is no item in `txArgs` array that is just an empty string. If that's the case, pls use my suggestion as it's much more readable.